minseo/service/userService.js

```diff
@@ -1,29 +1,52 @@
 const userDao = require('../models/userDao');
+const bcrypt = require('bcrypt');
 
-const signUp = async (name, email, password, profileImage) => {
+const jwt = require('jsonwebtoken');
+
+const signUp = async (name, email, password) => {
+  const saltRounds = 8;
   const pwValidation = new RegExp(
     '^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})'
   );
 
   const emailValidation = new RegExp(
-    '/^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i'
+    "([!#-'*+/-9=?A-Z^-~-]+(.[!#-'*+/-9=?A-Z^-~-]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")@([!#-'*+/-9=?A-Z^-~-]+(.[!#-'*+/-9=?A-Z^-~-]+)*|[[\t -Z^-~]*])"
   );
 
   if (!pwValidation.test(password) || !emailValidation.test(email)) {
     const err = new Error('PASSWORD_OR_EMAIL_IS_NOT_VALID');
-    err.statusCode = 400;
+    err.statusCode = 409;
     throw err;
   }
 
-  const createUser = await userDao.createUser(
-    name,
-    email,
-    password,
-    profileImage
-  );
+  const hashPassword = bcrypt.hashSync(password, saltRounds);
+
+  const createUser = await userDao.createUser(name, email, hashPassword);
   return createUser;
 };
 
+const signIn = async (email, password) => {
+  const user = await userDao.getUserByEmail(email);
+  const match = await bcrypt.compare(password, user.password);
+  if (email != user.email) {
+    const err = new Error('NOT_YOUR_EMAIL');
+    err.statusCode = 409;
+    throw err;
+  }
+  if (!match) {
+    const err = new Error('NOT_USER');
+    err.statusCode = 409;
+    throw err;
+  }
```

Comment:
1. email이 존재하지 않은 경우 user에는 빈 값이 들어갈 것입니다. 따라서 user.email을 비교하는 것이 아니고 user가 존재하는지 자체를 판단해야 됩니다.
2. user가 존재 하지 않으면 비밀번호를 확인 할 이유가 없기 때분에 로직순서는 아래와 같아야 합니다.
3. 인증과 관련된 Error이기 때문에 상태코드는 401을 사용해주세요
4. 이메일이 없을 때와 비밀번호가 틀렸을 때 에러 메시지는 통일 시켜주는 것이 좋습니다.
5. if가 시작되고 끝날 때 공백을 1칸씩 추가하여 가독성을 높여주세요.
```suggestion
  
  if (!user) {
    const err = new Error('INVALID_USER');
    err.statusCode = 401;
    throw err;
  }
  
  const isMatched = await bcrypt.compare(password, user.password);
  
  if (!isMatched) {
    const err = new Error('NOT_USER');
    err.statusCode = 401;
    throw err;
  }
```